bolt-sidecar/src/api/commitments/delegation/processor.rs

```diff
@@ -98,13 +128,20 @@ impl Future for CommitmentRequestProcessor {
                 this.pending_commitment_responses.poll_next_unpin(cx)
             {
                 progress = true;
+                let (id, response) = response;
+
                 let Ok(result_commitment) = response else {
                     error!("failed to receive commitment response. dropped sender");
                     continue;
                 };
 
                 if let Ok(commitment) = result_commitment {
                     trace!(?rpc_url, ?commitment, "received commitment response");
+                    let commitment = commitment
+                        .into_inclusion_commitment()
+                        .expect("no other commitment types")
```

Comment:
What do you think of using a match statement instead of this `into` function?
that way we will be prompted to update the code when we add a new enum variant